### PR TITLE
Sentry - Ignore exceptions when VBMS/VVA are in maintenance

### PR DESCRIPTION
**Pull request by aroltsch**
connects #763 

When VVA/VBMS is down, connect_vva and connect_vbms correctly catch the exception that is then passed to the eX and download gets a status of vva_connection_error or vbms_connection_error. 

When VBMS or VVA is in maintenance, do not capture these errors in Sentry.

### Note
Unfortunately, Sentry `excluded_exceptions` doesn't provide support for regular expressions so I created our own  class to exclude certain exceptions. 
See here what Raven does: https://github.com/getsentry/raven-ruby/blob/00c572b281b0862e2fbb7872b28c95e7c2578949/lib/raven/configuration.rb#L331

### CODE REVIEW

File: app/services/exception_logger.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class ExceptionLogger
+  DEPENDENCY_MAINTENANCE = [/Could not get JDBC Connection/, /Maintenance.+VBMS/, /upstream connect error or disconnect/].freeze
```
Comment (mdbenjam) [inline app/services/exception_logger.rb]: I'm wondering if it's easier to check the status code: 418? What do we think is more likely to change. Just a thought, I'll leave it up to you.

Comment (mdbenjam) [inline app/services/exception_logger.rb]: Does this check for just different instances of maintenance? Or also when they're down.

Comment (aroltsch) [reply app/services/exception_logger.rb]: So far we know of these 3 exceptions when VVA/VBMS are in maintenance. We can add more if we see more. I think VVA returns the same error when it is down or in maintenance.  I am not sure about VBMS.

File: app/services/manifest_fetcher.rb [hunk 2]
```diff
@@ -12,12 +12,7 @@ def process
     documents
   rescue *EXCEPTIONS => e
     manifest_source.update(status: :failed)
-    log_error(e)
+    ExceptionLogger.capture(e)
```
[the remark this answers is not in the document]
Comment (aroltsch) [reply app/services/manifest_fetcher.rb]: When a document download fails, we don't send any Sentry exceptions, we just set the status as failed.

**Pull request merged**